File: Problems/No.643_Maximum_Average_Subarray_I/maximum_average_subarray_I.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        
        // parameters
        double max_sum = INT_MIN;
        double partial_sum = 0;

        // start computing
        for (int i=0; i<nums.size(); i++) {

            partial_sum += nums[i];
            if (i < k-1) continue;
            
            max_sum = std::max(max_sum, partial_sum);
            partial_sum -= nums[i-k+1];

        }
        
        return max_sum/k;

    }
};

File: Problems/No.643_Maximum_Average_Subarray_I/maximum_average_subarray_I.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """

        # parameters
        max_sum, partial_sum = -float('inf'), 0.0

        # start computing
        for i in range(len(nums)):
            partial_sum += nums[i]
            if (i < k-1):
                continue
            
            max_sum = max(max_sum, partial_sum)
            partial_sum -= nums[i-k+1]
        
        return max_sum/k
